fvm/bootstrap.go

```diff
@@ -411,14 +413,55 @@ func (b *bootstrapExecutor) deployFungibleToken() flow.Address {
 	return fungibleToken
 }
 
-func (b *bootstrapExecutor) deployFlowToken(service, fungibleToken flow.Address) flow.Address {
+func (b *bootstrapExecutor) deployNonFungibleToken() flow.Address {
+	nonFungibleToken := b.createAccount(b.accountKeys.FungibleTokenAccountPublicKeys)
```

Comment:
this can break stuff with on emulator, changes flowtoken address.

Reply:
does it break it because the addresses are assigned in order? Is there any way around this?

Reply:
I guess we could create the FlowToken account first, then deploy the other contracts to other accounts, then deploy `FlowToken` after that to the first account created

Reply:
creating accounts in advance should make it backwards compatible I guess. https://github.com/onflow/flow-go/blob/7dda552c705866cc97c22846cb7e784d8017b3a0/fvm/bootstrap.go#L318-L326

Core contracts have fixed indexes https://github.com/onflow/flow-go/blob/91da6d0b24d1d5a5e156c3616f1179b4951d19a6/fvm/environment/account_creator.go#L15C1-L20C1 
better not to disturb them.

Reply:
you can create all accounts (like serviceAccount ), then reference them in the functions.

Reply:
something like this: 

```go
 service := b.createServiceAccount() 
 fungibleToken := b.deployFungibleToken() 

//create those to not change index 
 flowTokenAccount := b.createAccount(....)
 flowFeesAccount := b.createAccount(....)

 nonFungibleToken := b.deployNonFungibleToken() 
 b.deployMetadataViews(fungibleToken, nonFungibleToken) 
 flowToken := b.deployFlowToken(service, flowTokenAccount, fungibleToken, nonFungibleToken) 
 storageFees := b.deployStorageFees(service, fungibleToken, flowToken) 
 feeContract := b.deployFlowFees(service, flowFeesAccount, fungibleToken, flowToken, storageFees) 
  
 ```

Reply:
we can also deploy metadata views and non fungible tokens on the first/service account

Reply:
@janezpodhostnik actually this is much better, then we would be totally compatible with emulator.